theano/tensor/opt.py

```diff
@@ -1430,19 +1436,29 @@ def same_shape(self, x, y, dim_x=None, dim_y=None):
 
 class ShapeOptimizer(Optimizer):
     """Optimizer that serves to add ShapeFeature as an fgraph feature."""
-    def __init__(self):
-        Optimizer.__init__(self)
-
     def add_requirements(self, fgraph):
         fgraph.attach_feature(ShapeFeature())
 
     def apply(self, fgraph):
         pass
 
+
+class UnShapeOptimizer(Optimizer):
+    """Optimizer remove ShapeFeature as an fgraph feature."""
+    def apply(self, fgraph):
+        for feature in fgraph._features:
+            if isinstance(feature, ShapeFeature):
+                fgraph.remove_feature(feature)
```

Comment:
It was a metaphoric figure of speech saying that this looks like fragile code.

I really dislike using optimizers for all sorts of things that aren't optimizations and I would rather we spend some time figuring out a solution than pushing out a quick hack.

As for the old ones, see here: https://en.wikipedia.org/wiki/Old_One_in_fiction